### ISSUE TITLE
[Bug]: decompress_backfill fails with staging table not a hypertable error

### ISSUE THREAD

**Issue by murali89** [timescale/timescaledb#4472]
### What type of bug is this?

Unexpected error

### What subsystems and features are affected?

Distributed hypertable

## TL;DR
[`decompress_backfill`](https://docs.timescale.com/timescaledb/latest/how-to-guides/compression/backfill-historical-data/) with a *distributed hypertable* as the `destination_hypertable` fails with an unexpected error requiring staging table to be a hypertable: `ERROR: table "staging_table" is not a hypertable` (Full error details below).

# Reproduction Steps
 ## Create a staging table and sample data
 ```
 CREATE TABLE IF NOT exists staging_table (
	start_time timestamptz NOT NULL,
	request_id text NULL,
	account_uuid uuid NULL
);

INSERT INTO public.staging_table (start_time,request_id,account_uuid) VALUES
	 ('2022-03-05 05:47:35.449779-05','0011c42c-bace-410e-9194-574f893024a9','b384793a-16c3-4a4e-a5c3-0da0a587a913'),
	 ('2022-03-05 05:28:11.874011-05','001c55e8-f8ab-4b6b-8103-95bdbd1e653a','b384793a-16c3-4a4e-a5c3-0da0a587a913'),
	 ('2022-03-05 05:23:26.300802-05','00009f14-24d9-41e8-a14b-4c26e8749d61','b0fbcdb1-422c-42fc-ba50-a6d4bbef0782'),
	 ('2022-03-05 05:59:34.353057-05','001185c6-b331-4e5b-8d6f-bad2b028403d','b0fbcdb1-422c-42fc-ba50-a6d4bbef0782'),
	 ('2022-03-05 05:04:05.589386-05','000db254-161c-4c90-b8e2-b8e26fc42ae4','b0fbcdb1-422c-42fc-ba50-a6d4bbef0782'),
	 ('2022-03-05 05:52:20.956663-05','0018e5c4-d458-407c-9795-75a746aa842c','b0fbcdb1-422c-42fc-ba50-a6d4bbef0782'),
	 ('2022-03-05 05:25:28.621477-05','00122310-8947-4cfd-844a-5d9245744a80','b0fbcdb1-422c-42fc-ba50-a6d4bbef0782'),
	 ('2022-03-05 05:13:47.468799-05','001c4bdd-0cab-44f0-8ce8-7ee65c036fa0','b0fbcdb1-422c-42fc-ba50-a6d4bbef0782'),
	 ('2022-03-05 05:41:35.080609-05','001a3f4c-067b-4061-b48b-eaa2b74b0935','b0fbcdb1-422c-42fc-ba50-a6d4bbef0782'),
	 ('2022-03-05 05:28:02.181944-05','000fba7c-52e2-4670-9b2c-a2df277504d3','b0fbcdb1-422c-42fc-ba50-a6d4bbef0782');
 ```
 
 ## Create non-distributed hypertable and run decompress backfill (success case)
 ```
 CREATE TABLE dest_hypertable_non_dist (
	start_time timestamptz NOT NULL,
	request_id text NULL,
	account_uuid uuid NULL
)

select create_hypertable('dest_hypertable_non_dist', 'start_time');
CALL decompress_backfill(staging_table=>'staging_table', destination_hypertable=>'dest_hypertable_non_dist');
 ```
 
 * Runs successfully as expected with the following output:
   * `10 rows moved in range TIMESTAMPTZ '2022-03-05 05:04:05.589386-05' to TIMESTAMPTZ '2022-03-05 05:59:34.353058-05'`

## Create distributed hypertable and run decompress backfill (FAILURE CASE)
```
CREATE TABLE dest_hypertable_dist (
	start_time timestamptz NOT NULL,
	request_id text NULL,
	account_uuid uuid NULL
)

select create_distributed_hypertable('dest_hypertable_dist', 'start_time');

CALL decompress_backfill(staging_table=>'staging_table', destination_hypertable=>'dest_hypertable_dist');
```

* `decompress_backfill` call fails with below error:
```
SQL Error [TS001]: ERROR: table "staging_table" is not a hypertable
  Where: SQL statement "  
            WITH to_insert AS (DELETE 
            FROM staging_table --source table
            WHERE start_time >= NULL -- time column >= range start
            AND start_time < NULL -- time column < range end
            RETURNING * )
            INSERT INTO dest_hypertable_dist 
            SELECT * FROM to_insert
            ON CONFLICT DO NOTHING -- ON CONFLICT CLAUSE if it exists
            "
PL/pgSQL function decompress_backfill(regclass,regclass,text,boolean,interval,text[],boolean) line 149 at EXECUTE
```

### TimescaleDB version affected

 2.7.0

### PostgreSQL version used

psql (PostgreSQL) 14.4 (Ubuntu 14.4-1.pgdg20.04+1)

### What operating system did you use?

Ubuntu 20.04.4 LTS

### What installation method did you use?

Deb/Ubuntu

### What platform did you run on?

Other

### Relevant log output and stack trace

```bash
SQL Error [TS001]: ERROR: table "staging_table" is not a hypertable
  Where: SQL statement "  
            WITH to_insert AS (DELETE 
            FROM staging_table --source table
            WHERE start_time >= NULL -- time column >= range start
            AND start_time < NULL -- time column < range end
            RETURNING * )
            INSERT INTO dest_hypertable_dist 
            SELECT * FROM to_insert
            ON CONFLICT DO NOTHING -- ON CONFLICT CLAUSE if it exists
            "
PL/pgSQL function decompress_backfill(regclass,regclass,text,boolean,interval,text[],boolean) line 149 at EXECUTE
```

### How can we reproduce the bug?

```bash
Repro steps included in the description
```

**Comment by fabriziomello** [comment]
In [2.13.0](https://github.com/timescale/timescaledb/releases/tag/2.13.0) we announced the deprecation of multi-node and it will be complete removed from the upcoming 2.14.0.
